### ISSUE TITLE
Clean up path-resolution

### ISSUE THREAD

**Issue by fegorsch** [OGRECave/ogre#1696]
Currently the resolution of config-file paths (i.e. paths to `plugins.cfg` and `resources.cfg`) is complicated and thus hard to read/maintain and error-prone. Note that the following is only true for Windows, other OSs probably behave differently.

*standard-resolution for config-file paths*

- `FileSystemLayer::getConfigPaths` (private!): adds the executable's directory to `FileSystemLayer::mConfigPaths`
- `FileSystemLayer::getConfigFilePath`: looks for config files:
  - in `mHomePath` (a.k.a. the "writable path", `[home]/documents/[appname]` by default, can be changed with `setHomePath`)
  - in `mConfigPaths` (which contains the executable's directory, if `getConfigPaths` has been called before, can be changed with `setConfigPaths`)
  - if the file was not found there it assumes the config is located in the CWD and returns filename

*resources.cfg*

- `ApplicationContextBase::locateResources`: if `resources.cfg` is not found in one of the places from the standard-resolution, the "default media dir" (a.k.a. marco `OGRE_MEDIA_DIR`, i.e. `[installdir]/Media`) is set as the sole resource location.
- Side-note for completeness: `ApplicationContextBase::locateResources`: if `resources.cfg` was found in one of the places from the standard-resolution, the paths listed in `resources.cfg` are added as resource locations.

*plugins.cfg*

- if `plugins.cfg` is not found in one of the places from the standard-resolution, the macro `OGRE_CONFIG_DIR` (i.e. `[installdir]/bin/plugins.cfg`) is set as the location of `plugins.cfg`.
- Side-note for completeness: `Root::loadPlugins`: if envvar `OGRE_PLUGIN_DIR` does not exist, plugins are loaded from the `PluginFolder`-entry of `plugins.cfg`, otherwise from the value of `OGRE_PLUGIN_DIR`.

*tl;dr*

The search order for config files is:

1. "writable path" (`[home]/documents/[appname]` or what was set with `setHomePath`)
2. "config paths" (executable's dir or what was set with `setConfigPaths`)
3. CWD
4. installdir

- Check if this search order makes sense.
- Check if the resolution can be simplified. E.g. add all search paths to `mConfigPaths` in the desired search order.

Keep in mind, that this description refers to Windows only and needs to be verified for all other OSs.
